pkg/apis/config/v1alpha1/resmgr/policy/balloons/config.go

```diff
@@ -149,6 +149,10 @@ type BalloonDef struct {
 	// this will be the number of CPUs reserved for it even if a container
 	// would request less.
 	MinCpus int `json:"minCPUs,omitempty"`
+	// MemoryTypes lists memory types allowed to containers in a
+	// balloon. Supported types are: DRAM, HBM, PMEM. By default
+	// all memory types in the system are allowed.
+	MemoryTypes []string `json:"memoryTypes,omitempty"`
```

Comment:
I wonder if something like this would provide working extended validation for this:

```golang
type BalloonDef struct {
...
        // MemoryTypes lists memory types allowed to containers in a
        // balloon. Supported types are: DRAM, HBM, PMEM. By default
        // all memory types in the system are allowed.
        // +kubebuilder:validation:XValidation:rule="self.values.all(value == 'DRAM' || value == 'HBM' || value == 'PMEM')", message="invalid memoryTypes value, should be one of DRAM, HBM, or PMEM"
	MemoryTypes []string `json:"memoryTypes,omitempty"`
...
}
```

Reply:
Okay, so we'll need this after all (since CEL-expressions are just unusable for this as we found out due to very limited expression evaluation runtime budget which is getting exceeded here due to double nested slices):
```golang
...
    // +kubebuilder:validation:items:Pattern=^(DRAM|HBM|PMEM)$
    MemoryTypes []string `json:"memoryTypes,omitempty"`
...
```

Reply:
Thank you, @klihub! This has now a nice solution and we did not consume CEL budget for it.